analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -967,6 +967,21 @@ public void Invocation_ObjectEquals_LearnResult(string left, string right, bool
         validator.TagValue("Right").AllConstraints.Should().ContainSingle().Which.Kind.Should().Be(expectedConstraintRight);
     }
 
+    [DataTestMethod]
+    [DataRow("true", "null", false)]
+    [DataRow("null", "true", false)]
+    [DataRow("null", "(bool?)true", false)]
+    public void Invocation_ObjectEqualsMixedNullObjectArguments_LearnResult(string left, string right, bool expectedResult)
```

Comment:
I think you can just incorporate this into `Invocation_BoolEquals_LearnsResult`.

Reply:
It calls ObjectEquals and tests this method's behavior :/

Reply:
Maybe just add it to `Invocation_ObjectEquals_LearnResult` then.

Reply:
It's not straightforward - as discussed will leave as is.